### DISCUSSION TITLE
State D (ventilation required)

### DISCUSSION THREAD

**Discussion by nagyrobi** [dzurikmiroslav/esp32-evse#53] [General]
Congrats to this nice project!

A quick question: how does this unit handle the car's state D (ventilation required) signalization?

According to the standard, outdoors chargers should continue charging. Indoors (like garage) chargers should trigger a ventilation system of the area and only continue charging when that works.

In reality, most commercially chargers simply stop charging the vehicle when state D is signalled. And don't really see any option to control an external relay or similar, to enable building ventilation. Or a simple override option for when the vehicle is being charged outdoors (eg. at the side of the house).

**Comment by dzurikmiroslav** [comment]
Hi thanks :)

Currently state D is handled as state C. I thought about making a configuration option, that would allow charging in state D or go to state E. 

For indoor installation, there is Lua scripting. You can writte a script which activate ventilation system in state D (low/hight side output).

**Reply by nagyrobi** [reply]
Didn't check, is there a free GPO to hook a relay onto for ventilation?

**Reply by dzurikmiroslav** [reply]
Yes ESP32-S2 EVSE DIY ALPHA board has OUT1-4 fully controllable from Lua
